app/src/main/java/com/nervousfish/nervousfish/activities/ChangeContactActivity.java

```diff
@@ -67,11 +73,23 @@ public void saveContact(final View v) {
         final InputMethodManager imm = (InputMethodManager) getSystemService(INPUT_METHOD_SERVICE);
         imm.hideSoftInputFromWindow(v.getWindowToken(), 0);
 
-        final EditText editText = (EditText) findViewById(R.id.edit_contact_name_input);
-        if (isValidName(editText.getText().toString())) {
+        final EditText editTextName = (EditText) findViewById(R.id.edit_contact_name_input);
+        final EditText editTextIBAN = (EditText) findViewById(R.id.contact_page_change_iban);
+        String ibanString = editTextIBAN.getText().toString();
+        final boolean validName = isValidName(editTextName.getText().toString());
+        boolean validIban = IBANVerifier.isValidIBAN(ibanString);
+        if (!validIban && (EMPTY_STRING.equals(ibanString) || "-".equals(ibanString))) {
+            validIban = true;
+            ibanString = EMPTY_STRING;
+        }
+        if (validName && validIban) {
             //Update contact
             try {
-                final Contact newContact = new Contact(editText.getText().toString(), contact.getKeys());
+                Contact newContact = new Contact(editTextName.getText().toString(), contact.getKeys());
+                if (!ibanString.equals(EMPTY_STRING)) {
```

Comment:
Judging from the `CreateProfileActivity`, the IBan **is** required. Why is it not required here?